src/signature/src/main.rs

```diff
@@ -1,14 +1,26 @@
+use candid::{CandidType, Decode, Deserialize, Encode, Principal};
 use ethers::prelude::*;
+use ic_agent::agent::http_transport::ReqwestHttpReplicaV2Transport;
+use ic_agent::identity::AnonymousIdentity;
+use ic_agent::{Agent, AgentError};
+
+#[derive(Debug, Clone, CandidType, Deserialize)]
+struct RegistrationInfo {
+    minter_address: String,
+    registration_fee: u64,
+}
 
 #[tokio::main]
 async fn main() {
+    let minter_address = get_minter_address().await.expect("call evmc error");
+
     let wallet = LocalWallet::new(&mut rand::thread_rng());
     println!("private key: {:?}", wallet.signer().to_bytes().to_vec());
 
     // ======== generate evm registry tx ========
     let tx = TransactionRequest::new()
         .from(wallet.address())
-        .to("0xb0e5863d0ddf7e105e409fee0ecc0123a362e14b") // MINTER_ADDRESS
+        .to(minter_address.as_str()) // MINTER_ADDRESS
         .value(100_000) // REGISTRATION_FEE
```

Comment:
also the registration fee has to be retrieved from the endpoint 😅

Reply:
fixed